app/Http/Controllers/Webapp/PostController.php

```diff
@@ -0,0 +1,1890 @@
+            ]))->send($action)->getResult();
+            $post->content = Post::replaceIDButtonLink($returnData->data->post_id, $post->content, $post->target_post_id);
+            $post->target_post_id = $returnData->data->post_id;
+            $post->published_status = 1;
+            $this->_flashMessage($request, 'OK', 'success');
+
+            $this->activityLog->addMeta('TargetPostId', $post->target_post_id);
+        } catch(\Exception $e) {
+            $post->content = Post::replaceIDButtonLink('__POSTID__', $post->content, $post->target_post_id);
+            $this->activityLog->setInfo('Error when reaching to target site');
+            $this->activityLog->addMeta('FailedPostToTargetSite', 'error when reaching to target site');
+            $this->_flashMessage($request, '更新に失敗しました。もう一度ご確認ください。', 'warning');
+        }
+
+        $post->save();
+
+        $this->activityLog->addMeta('NewPublishedStatus', $post->published_status);
+
+        $next = action('Webapp\\PostController@getIndex');
+
+        if(($from = $request->get('from', null) != null) && $from == 'preview') {
+            $next = $post->getTargetSiteUrl();
+            $this->_clearFlashMessage($request);
+        }
+
+        if($request->get('hasFilter', false) !== false) {
+            return redirect()->back()->send();
+        }
+
+        if(!$request->get('selected_ids'))
+        return redirect()->to($next)->send();
+    }
+
+    public function getDraft(Request $request, $postId = 0) {
+        $this->activityLog->setPostId($postId);
+
+        if(empty($post = Post::find($postId))) {
+            $this->activityLog->setInfo('Not found');
+            return '404 Not Found';
+        }
+
+        if(!$post->canDraftBy($this->currentUser)) {
+            $this->_flashMessage($request, 'Draft is not permitted', 'danger');
+            $this->activityLog->setInfo('Not permitted');
+            return redirect()->to(action('Webapp\\PostController@getIndex'))->send();
+        }
+
+        if($post->isLock($this->currentSession)) {
+            $this->_flashMessage($request, 'このコンテンツは現在ロックされています。下書きに保存できません。', 'danger');
+            $this->activityLog->setInfo('Locked');
+            if(!$request->get('selected_ids'))
+                return redirect()->to(action('Webapp\\PostController@getIndex'))->send();
+            else
+                return;
+        }
+        if($post->site_id == ''){
+            $this->_flashMessage($request, 'ターゲットサイトを選んでください', 'danger');
+            $this->activityLog->setInfo('Missing target stie');
+            return redirect()->to(action('Webapp\\PostController@getIndex'))->send();
+        }
+
+        if($post->published_status == 0) {
+            $this->_flashMessage($request, '下書きに保存しました。', 'warning');
+            $this->activityLog->setInfo('Already draft');
+            if(!$request->get('selected_ids'))
+                return redirect()->to(action('Webapp\\PostController@getIndex'))->send();
+            else
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.relipa.blog/coding_rules#rule_key=php%3AS121)